pilot/pkg/model/telemetry.go

```diff
@@ -1052,3 +1053,51 @@ func generateStatsConfig(class networking.ListenerClass, metricsCfg telemetryFil
 func disableHostHeaderFallback(class networking.ListenerClass) bool {
 	return class == networking.ListenerClassSidecarInbound || class == networking.ListenerClassGateway
 }
+
+type OTelResourceSemantic string
+
+const (
+	KubernetesCluster   OTelResourceSemantic = "k8s.cluster.name"
```

Comment:
Doesn't otlp define those constants ? I don't usually ask for more dependencies, but they use go.mod to keep them small, and it would make it clear we use the same names - and if we use any name not defined by their semantic convention it'll be easier to flag.

I agree with John - it must match baggage header processing, and it needs to be replicated to the ambient branch as well.

Reply:
@costinm OTLP uses generic strings as keys, it's not schematized.

Baggage does seem to follow apart from the canonical stuff. Not sure how the vendor extensions are meant to be represented. We probably need a canonical document somewhere, not in the code.

VMs shouldn't be represented as k8s, use https://opentelemetry.io/docs/reference/specification/resource/semantic_conventions/host/ instead.